src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/MissingNullCaseInSwitchCheck.java

```diff
@@ -0,0 +1,210 @@
+ * as there is no need for an external null check before entering the switch.
+ * </p>
+ * <p>
+ * Limitations due to not being type-aware:
+ * </p>
+ * <ul>
+ * <li>
+ * This check does not differentiate between primitive types and
+ * their boxed counterparts in case labels.
+ * Therefore, any case label with an integer will not violate this check,
+ * even if it is a reference type, i.e. {@code Integer}.
+ * </li>
+ * <li>
+ * This check does not support enums; syntactically, these are no different from other constants.
+ * </li>
+ * </ul>
+ * <p>
+ * See the <a href="https://docs.oracle.com/javase/specs/jls/se22/html/jls-15.html#jls-15.28">
+ * Java Language Specification</a> for more information about switch statements and expressions.
+ * </p>
+ * <p>
+ * Parent is {@code com.puppycrawl.tools.checkstyle.TreeWalker}
+ * </p>
+ * <p>
+ * Violation Message Keys:
+ * </p>
+ * <ul>
+ * <li>
+ * {@code missing.switch.nullcase}
+ * </li>
+ * </ul>
+ *
+ * @since 10.18.0
+ */
+
+@StatelessCheck
+public class MissingNullCaseInSwitchCheck extends AbstractCheck {
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_KEY = "missing.switch.nullcase";
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return new int[] {TokenTypes.LITERAL_SWITCH};
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        final List<DetailAST> caseLabels = getAllCaseLabels(ast);
+        final boolean hasNullCaseLabel = caseLabels.stream()
+                .anyMatch(MissingNullCaseInSwitchCheck::hasLiteralNull);
+        if (!hasNullCaseLabel) {
+            final boolean hasPatternCaseLabel = caseLabels.stream()
+                .anyMatch(MissingNullCaseInSwitchCheck::hasPatternCaseLabel);
+            final boolean hasStringCaseLabel = caseLabels.stream()
+                .anyMatch(MissingNullCaseInSwitchCheck::hasStringCaseLabel);
+            if (hasPatternCaseLabel || hasStringCaseLabel) {
+                log(ast, MSG_KEY);
+            }
+        }
+    }
+
+    /**
+     * Gets all case labels in the given switch AST node.
+     *
+     * @param detailAST the AST node representing {@code LITERAL_SWITCH}
+     * @return a list of all case labels in the switch
+     */
+    private static List<DetailAST> getAllCaseLabels(DetailAST detailAST) {
+        final List<DetailAST> caseLabels = new ArrayList<>();
+        DetailAST ast = detailAST.getFirstChild();
+        while (ast != null) {
+            // case group token may have several LITERAL_CASE tokens
+            TokenUtil.forEachChild(ast, TokenTypes.LITERAL_CASE, caseLabels::add);
+            ast = ast.getNextSibling();
+        }
+        return caseLabels;
```

Comment:
Please return an immutable list.

Reply:
done

Reply:
```
+  <mutation unstable="false">
+    <sourceFile>MissingNullCaseInSwitchCheck.java</sourceFile>
+    <mutatedClass>com.puppycrawl.tools.checkstyle.checks.coding.MissingNullCaseInSwitchCheck</mutatedClass>
+    <mutatedMethod>getAllCaseLabels</mutatedMethod>
+    <mutator>org.pitest.mutationtest.engine.gregor.mutators.experimental.ArgumentPropagationMutator</mutator>
+    <description>replaced call to java/util/Collections::unmodifiableList with argument</description>
+    <lineContent>return Collections.unmodifiableList(caseLabels);</lineContent>
+  </mutation>
```

I thought this was an issue with Pitest and solved by https://github.com/checkstyle/checkstyle/pull/14484 ?

Reply:
seems like this is not activated for the pitest-coding-2 profile yet. we can suppress for now ?

Reply:
I think the suppression would be more LOC than just adding the feature to this profile, right?

Reply:
right, done